### PR TITLE
Handle multiple textures in protobuf.

### PR DESCRIPTION
**Pull request by gaschler**
Handle the protobuf change from
https://github.com/googlecartographer/cartographer/pull/519
by forwarding the first SubmapTexture if multiple are available.

### CODE REVIEW

File: cartographer_ros/cartographer_ros/map_builder_bridge.cc [hunk 1]
```diff
@@ -105,13 +105,20 @@ bool MapBuilderBridge::HandleSubmapQuery(
   }
 
   response.submap_version = response_proto.submap_version();
-  response.cells.insert(response.cells.begin(), response_proto.cells().begin(),
-                        response_proto.cells().end());
-  response.width = response_proto.width();
-  response.height = response_proto.height();
-  response.resolution = response_proto.resolution();
+  if (response_proto.textures_size() == 0) {
+    LOG(WARNING) << "Empty textures given.";
```
[the remark this answers is not in the document]
Comment (gaschler) [reply cartographer_ros/cartographer_ros/map_builder_bridge.cc]: Done

File: cartographer_ros/cartographer_ros/map_builder_bridge.cc [hunk 2]
```diff
@@ -105,13 +105,20 @@ bool MapBuilderBridge::HandleSubmapQuery(
   }
 
   response.submap_version = response_proto.submap_version();
-  response.cells.insert(response.cells.begin(), response_proto.cells().begin(),
-                        response_proto.cells().end());
-  response.width = response_proto.width();
-  response.height = response_proto.height();
-  response.resolution = response_proto.resolution();
+  if (response_proto.textures_size() == 0) {
```
Comment (wohe) [inline cartographer_ros/cartographer_ros/map_builder_bridge.cc]: Sorry, did not realize this is answering a query, not receiving an empty answer:
```
CHECK(response_proto.textures_size() > 0) << ...
```
And include the ID in the error message to aid debugging should this crash?

Comment (gaschler) [reply cartographer_ros/cartographer_ros/map_builder_bridge.cc]: Done.

File: cartographer_ros/cartographer_ros/map_builder_bridge.cc [hunk 3]
```diff
@@ -105,10 +105,9 @@ bool MapBuilderBridge::HandleSubmapQuery(
   }
 
   response.submap_version = response_proto.submap_version();
-  if (response_proto.textures_size() == 0) {
-    LOG(ERROR) << "Empty textures given.";
-    return false;
-  }
+  CHECK(response_proto.textures_size() > 0)
+      << "empty textures given for trajectory_id " << request.trajectory_id
```
Comment (wohe) [inline cartographer_ros/cartographer_ros/map_builder_bridge.cc]: nit: There is an operator<< for `SubmapId`.
```
<< "Empty textures given for submap: " << submap_id
```
and extract `submap_id` above into a local variable?

Comment (gaschler) [reply cartographer_ros/cartographer_ros/map_builder_bridge.cc]: Done, PTAL.

**Comment by gaschler** [comment]
@wohe could you PTAL so we can sync this with the protobuf change in googlecartographer/cartographer#519 ?

**Comment by wohe** [comment]
No need to sync anything since this is another repo. I already restarted the CI. Just takes a while.

**Pull request merged**